node/src/runner.rs

```diff
@@ -105,17 +98,81 @@ pub async fn run(options: Options) -> Result<()> {
                 .with_context(|| format!("Failed to write config to the '{path:?}' file"))?;
             Ok(())
         }
-        Command::Run(ref run_options) => {
-            let node_config = NodeConfig::read(&options.config_path(), run_options)
-                .context("Failed to initialize config")?;
-            log::trace!("Starting with the following config\n: {node_config:#?}");
-            let manager = initialize(run_options.net, node_config).await?;
-            #[allow(clippy::unit_arg)]
-            Ok(manager.main().await)
+        Command::Mainnet(ref run_options) => {
+            let chain_config = common::chain::config::create_mainnet();
+            start(&options.config_path(), run_options, chain_config).await
+        }
+        Command::Regtest(ref regtest_options) => {
+            let chain_config = regtest_chain_config(&regtest_options.chain_config)?;
+            start(
+                &options.config_path(),
+                &regtest_options.run_options,
+                chain_config,
+            )
+            .await
         }
     }
 }
 
+async fn start(
+    config_path: &Path,
+    run_options: &RunOptions,
+    chain_config: ChainConfig,
+) -> Result<()> {
+    let node_config =
+        NodeConfig::read(config_path, run_options).context("Failed to initialize config")?;
+    log::trace!("Starting with the following config\n: {node_config:#?}");
```

Comment:
I would make this info, not trace.

Reply:
I see a newline in the log message. How does that render in the output?

Reply:
@iljakuklic Actually there are many new lines here, because the `{:#?}` formatting produces a "pretty" output by splitting it into multiple lines:
```
> RUST_LOG=info ./node -d ./eee mainnet
[2022-07-25T14:58:36Z INFO  node::runner] Starting with the following config:
     NodeConfig {
        datadir: "./eee",
        chainstate: ChainstateConfig {
            max_db_commit_attempts: 10,
            max_orphan_blocks: 512,
        },
        p2p: P2pConfig {
            bind_address: "/ip6/::1/tcp/3031",
            ban_threshold: 100,
            outbound_connection_timeout: 10,
        },
        rpc: RpcConfig {
            bind_address: 127.0.0.1:3030,
        },
    }
[2022-07-25T14:58:36Z INFO  subsystem::manager] Initializing subsystem manager mintlayer
...
```
What do you think? Should I remove the new lines from the output?

Reply:
Not sure. I was mainly concerned it would continue at the start of the line. However I see it's indented so should be fine. Still makes grepping a bit annoying. So at this point it's mainly aesthetic preference and I don't mind too much either way. Let's see what others think.

Reply:
Would be nice if we could convert this to a table... it's a typical thing in command line arguments to print a table on start. I can live with this for now though.